python/lsst/ts/eas/m1m3ts_model.py

```diff
@@ -0,0 +1,277 @@
+        self.heater_setpoint_delta = heater_setpoint_delta
+        self.m1m3_setpoint_cadence = m1m3_setpoint_cadence
+        self.setpoint_deadband_heating = setpoint_deadband_heating
+        self.setpoint_deadband_cooling = setpoint_deadband_cooling
+        self.maximum_heating_rate = maximum_heating_rate
+        self.features_to_disable = features_to_disable
+
+        # Last setpoint, for deadband purposes
+        self.last_m1m3ts_setpoint: float | None = None
+
+    async def monitor(self) -> None:
+        self.log.debug("M1M3TSModel.monitor")
+
+        async with salobj.Remote(
+            domain=self.domain,
+            name="MTM1M3TS",
+        ) as m1m3ts_remote:
+            if "m1m3ts" not in self.features_to_disable:
+                m1m3ts_future = asyncio.gather(
+                    self.follow_ess_indoor(m1m3ts_remote=m1m3ts_remote),
+                    self.wait_for_noon(m1m3ts_remote=m1m3ts_remote),
+                )
+
+                self.log.debug("M1M3TSModel.monitor closing...")
+                try:
+                    await m1m3ts_future
+                except asyncio.CancelledError:
+                    m1m3ts_future.cancel()
+                    await asyncio.gather(m1m3ts_future, return_exceptions=True)
+                    raise
+
+            else:
+                # If m1m3ts is disabled, just sleep.
+                while True:
+                    try:
+                        await asyncio.sleep(DORMANT_TIME)
+                    except asyncio.CancelledError:
+                        self.log.info("monitor cancelled")
+                        raise
+
+    async def apply_setpoints(
+        self, *, m1m3ts_remote: salobj.Remote, setpoint: float
+    ) -> None:
+        glycol_setpoint = setpoint + self.glycol_setpoint_delta
+        heaters_setpoint = setpoint + self.heater_setpoint_delta
+        self.log.debug(f"Setting MTM1MTS: {glycol_setpoint=} {heaters_setpoint=}")
+        if hasattr(m1m3ts_remote, "cmd_applySetpoints"):
+            await m1m3ts_remote.cmd_applySetpoints.set_start(
+                glycolSetpoint=glycol_setpoint,
+                heatersSetpoint=heaters_setpoint,
+            )
+        else:
+            await m1m3ts_remote.cmd_applySetpoint.set_start(
+                glycolSetpoint=glycol_setpoint,
+                heatersSetpoint=heaters_setpoint,
+            )
+
+    async def follow_ess_indoor(self, *, m1m3ts_remote: salobj.Remote) -> None:
+        self.log.debug("follow_ess_indoor")
+
+        async with salobj.Remote(
+            domain=self.domain,
+            name="ESS",
+            index=self.indoor_ess_index,
+            include=("temperature",),
+        ) as ess_remote:
+            while True:
+                if self.dome_model.dome_is_closed:
+                    await asyncio.sleep(DORMANT_TIME)
+                    continue
+
+                temperatures = []
+                start_time = utils.current_tai()
```

Comment:
make it end_time = utils.current_tai() + self.m1m3_setpoint_cadence, and change test to current_tai() < end_time.

Or even better, instead of TAI which is "expansive" to compute, use time.monotonic() to get monotonic clock.